src/CategoricalArrays.jl

```diff
@@ -13,6 +13,11 @@ module CategoricalArrays
     using Missings
     using Printf
 
+    # JuliaLang/julia#36810 
+    if VERSION < v"1.6.0"
```

Comment:
This should be enough (assuming @nalimilan does not want us to be more precise :smile:)
```suggestion
    if VERSION < v"1.5.1"
```